cmd/flags.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var generateReadmeFlags = map[string]bool{
+	"print": true,
+}
+
+func newListFlagsCommand(rootCmd *cobra.Command) *cobra.Command {
+	return &cobra.Command{
+		Use:   "list-flags",
+		Short: "Lists all flags for documentation",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			printCommandFlags(rootCmd, "")
+			return nil
+		},
+	}
+}
+
+func printCommandFlags(cmd *cobra.Command, indent string) {
+	_, ok := generateReadmeFlags[cmd.Use]
+	if ok {
+		fmt.Printf("%s### `%s` command\n\n", indent, cmd.Use)
+
+		fmt.Printf("%s| Flag                          | Default Value            | Required | Description                                                  |\n", indent)
+		fmt.Printf("%s| ----------------------------- | ------------------------ | -------- | ------------------------------------------------------------ |\n", indent)
```

Comment:
This hardcoded tabulation is a bit odd, as in case we introduce a flag named `very-very-very-very-long-flag`, the whole table messes up. Can we use some package like github.com/olekukonko/tablewriter as we already do for the [notifications](https://github.com/kubernetes-sigs/ingress2gateway/blob/beb19dc871bce8c47a114563ce72f0ea69e869d2/pkg/i2gw/notifications/notifications.go#L69) package?